### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Daniil-Pankieiev**

### CODE REVIEW

**Review by RVChornyy — APPROVED** [review]

File: cinema/serializers.py [hunk 1]
```diff
@@ -1,24 +1,111 @@
...
+    actors = serializers.PrimaryKeyRelatedField(
+        many=True,
+        queryset=Actor.objects.all(),
+        required=False
+    )
+    genres = serializers.PrimaryKeyRelatedField(
+        many=True,
+        queryset=Genre.objects.all(),
+        required=False
+    )
```
Comment (wQuelS) [inline cinema/serializers.py]: you were not supposed to do so in **_This_** task according to the description
image

File: cinema/models.py [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    def __str__(self) -> str:
+        return self.name
+
+    @property
+    def capacity(self) -> int:
+        return self.rows * self.seats_in_row
```
Comment (wQuelS) [inline cinema/models.py]: do you really need this?))
Nothing wrong yet, but such implementation may cause additional questions

File: cinema/serializers.py [hunk 3]
```diff
@@ -1,24 +1,111 @@
...
+        instance.first_name = validated_data.get(
+            "first_name",
+            instance.first_name
+        )
+        instance.last_name = validated_data.get(
+            "last_name",
+            instance.last_name
+        )
+        instance.save()
+        return instance
```
Comment (wQuelS) [inline cinema/serializers.py]: this code ( as well as similar code in the next serializers) will not be needed if you do everything according to the task description. Fix everywhere

File: cinema/urls.py [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actor-detail"),
+    path("cinema_halls/", cinemahall_list, name="cinemahall-list"),
+    path(
+        "cinema_halls/<int:pk>/",
+        cinemahall_detail,
+        name="cinemahall-detail"
+    ),
+    path("", include(router.urls))
```
Comment (wQuelS) [inline cinema/urls.py]: this is not a reason for reject, however such general path is better to place at the beginning:)

File: cinema/views.py [hunk 5]
```diff
@@ -1,45 +1,113 @@
...
+class CinemaHallViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
```
Comment (wQuelS) [inline cinema/views.py:115]: Did you forget about MoviViewSet?

**Review by wQuelS — CHANGES_REQUESTED** [review]

**Review by wQuelS — APPROVED** [review]
GJ

**Pull request closed without merging**